brave-propagation-aws/src/main/java/brave/propagation/aws/AWSExtractor.java

```diff
@@ -0,0 +1,202 @@
+          op = Op.SKIP;
+        } else {
+          op = Op.EXTRA;
+          if (extraFields == null) extraFields = new StringBuilder();
+          extraFields.append(';').append(currentString);
+        }
+        currentString.setLength(0);
+      } else if (op == null) {
+        currentString.append(c);
+        continue;
+      }
+      // no longer whitespace
+      switch (op) {
+        case EXTRA:
+          extraFields.append(c);
+          while (i < length && (c = traceIdString.charAt(i)) != ';') {
+            extraFields.append(c);
+            i++;
+          }
+          break;
+        case SKIP:
+          while (++i < length && traceIdString.charAt(i) != ';') {
+            // skip until we hit a delimiter
+          }
+          break;
+        case ROOT:
+          if (i + ROOT_LENGTH > length // 35 = length of 1-67891233-abcdef012345678912345678
+              || traceIdString.charAt(i++) != '1'
+              || traceIdString.charAt(i++) != '-') {
+            break OUTER; // invalid version or format
+          }
+          // Parse the epoch seconds and high 32 of the 96 bit trace ID into traceID high
+          for (int hyphenIndex = i + 8, endIndex = hyphenIndex + 1 + 8; i < endIndex; i++) {
+            c = traceIdString.charAt(i);
+            if (c == '-' && i == hyphenIndex) continue; // skip delimiter between epoch and random
+            traceIdHigh <<= 4;
+            if (c >= '0' && c <= '9') {
+              traceIdHigh |= c - '0';
+            } else if (c >= 'a' && c <= 'f') {
+              traceIdHigh |= c - 'a' + 10;
+            } else {
+              break OUTER; // invalid format
+            }
+          }
+          // Parse the low 64 of the 96 bit trace ID into traceId
+          for (int endIndex = i + 16; i < endIndex; i++) {
+            c = traceIdString.charAt(i);
+            traceId <<= 4;
+            if (c >= '0' && c <= '9') {
+              traceId |= c - '0';
+            } else if (c >= 'a' && c <= 'f') {
+              traceId |= c - 'a' + 10;
+            } else {
+              break OUTER; // invalid format
+            }
+          }
+          break;
+        case PARENT:
+          long parentId = 0L;
+          for (int endIndex = i + 16; i < endIndex; i++) {
+            c = traceIdString.charAt(i);
+            parentId <<= 4;
+            if (c >= '0' && c <= '9') {
+              parentId |= c - '0';
+            } else if (c >= 'a' && c <= 'f') {
+              parentId |= c - 'a' + 10;
+            } else {
+              break OUTER; // invalid format
+            }
+          }
+          parent = parentId;
+          break;
+        case SAMPLED:
+          c = traceIdString.charAt(i++);
+          if (c == '1') {
+            sampled = true;
+          } else if (c == '0') {
+            sampled = false;
+          }
+          break;
+      }
+      op = null;
+    }
+
+    AmznTraceId amznTraceId = EXTRA_MARKER;
```

Comment:
only difference is this where we look for the specific marker as opposed to making a `List<Object>` for extra